Src/jpdf.cpp

```diff
@@ -445,12 +451,18 @@ main (int   argc,
                         const FArrayBox &myFab = (*mf[iLevel])[ntmfi];
                         const Real *dx = amrData.DxLevel()[iLevel].dataPtr();
                         const Real *cPtr  = myFab.dataPtr(cVar); // Conditioning
-                        const Real *v1Ptr = myFab.dataPtr(var1);
+                        const Real *v1Ptr = myFab.dataPtr(var1); 
                         const Real *v2Ptr = myFab.dataPtr(var2);
-                        const Real *isPtr = myFab.dataPtr(nVars); // Intersect
-			const Box&  bx    = ntmfi.validbox();
-			const int  *lo    = bx.loVect();
-			const int  *hi    = bx.hiVect(); 
+                        Vector<const Real*> condVarPtrs(nVars_condMean);
+                        for (int v=0; v<nVars_condMean; v++) {
+                            condVarPtrs[v] = myFab.dataPtr(ix_condMean[v]);
+                        }
+                        const Real *isPtr = myFab.dataPtr(ix_isVar); // Intersect
+
+                        const Box&  bx    = ntmfi.validbox();
+                        const int  *lo    = bx.loVect();
```

Comment:
I would prefer that from this point onwards, we could switch to the newer syntax. There will be some subtleties for getting it working with the GPUs, but we can sort this out when we modernise the other tools. Take a look at some of the newer tools to get a feel for how to do the ijk loops. It will shorten the next 60 lines or so.